===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ rec_encrypt:n$65 == { 0 } }) free(*(rec_encrypt:real_in)) at 63 (line 115, column 17) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/recutils-1.8/src/rec-crypt.c
+++ b/table1/benchmarks/recutils-1.8/src/rec-crypt.c
@@ -112,7 +112,10 @@ rec_encrypt (char   *in,
   if (padding != 0)
     {
       real_in_size = real_in_size + padding;
-      real_in = realloc (real_in, real_in_size);
+      tmp = realloc(real_in, real_in_size);
+      if (tmp == NULL)
+          free(real_in);
+      real_in = tmp;
 
       for (i = 0; i < padding; i++)
         real_in[real_in_size - i - 1] = '\0';
